src/endpoint/s3/ops/s3_head_object.js

```diff
@@ -11,6 +11,7 @@ const S3Error = require('../s3_errors').S3Error;
  */
 async function head_object(req, res) {
     const encryption = s3_utils.parse_encryption(req);
+    s3_utils.parse_version_id(req.query.versionId);
```

Comment:
also grab the returned value here and replace refs to req.query.versionId in this function
```suggestion
    const version_id = s3_utils.parse_version_id(req.query.versionId);
```